packages/fxa-content-server/app/scripts/models/auth_brokers/oauth-webchannel-v1.js

```diff
@@ -0,0 +1,117 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * WebChannel OAuth broker that speaks 'v1' of the protocol.
+ */
+import _ from 'underscore';
+import ChannelMixin from './mixins/channel';
+import Cocktail from 'cocktail';
+import Constants from '../../lib/constants';
+import HaltBehavior from '../../views/behaviors/halt';
+import OAuthRedirectAuthenticationBroker from './oauth-redirect';
+import ScopedKeys from 'lib/crypto/scoped-keys';
+import WebChannel from '../../lib/channels/web';
+import SyncEngines from '../sync-engines';
+
+const proto = OAuthRedirectAuthenticationBroker.prototype;
+
+const OAuthWebChannelBroker = OAuthRedirectAuthenticationBroker.extend({
+  defaultBehaviors: _.extend({}, proto.defaultBehaviors, {
+    afterForceAuth: new HaltBehavior(),
+    afterSignIn: new HaltBehavior(),
+  }),
+
+  defaultCapabilities: _.extend({}, proto.defaultCapabilities, {
+    chooseWhatToSyncWebV1: true,
+    fxaStatus: true,
+    openWebmailButtonVisible: false,
+  }),
+
+  commands: _.pick(WebChannel, 'FXA_STATUS', 'OAUTH_LOGIN'),
+
+  type: 'oauth-webchannel-v1',
+
+  initialize(options = {}) {
+    this.session = options.session;
+    this._channel = options.channel;
+    this._scopedKeys = ScopedKeys;
+    this._metrics = options.metrics;
+
+    proto.initialize.call(this, options);
+    this.request(this.getCommand('FXA_STATUS'), {
+      service: this.relier.get('service'),
+    }).then(response => this.onFxaStatus(response));
+  },
+
+  /**
+   * Handle a response to the `fxa_status` message.
+   *
+   * @param {any} [response={}]
+   * @private
+   */
+  onFxaStatus(response = {}) {
+    const supportedEngines =
+      response.capabilities && response.capabilities.engines;
+    if (supportedEngines) {
+      // supportedEngines override the defaults
```

Comment:
This is a subtle change in behaviour from the desktop broker, I think it would be worth calling out more explicitly in e.g. the new protocol doc for the oauth-webchannel broker.

Reply:
(Thought I answered this somewhere already) Yes, the native app like Fenix will send in the engines in supports